### PR TITLE
RUN-1968 fix: sanitize workload metadata

### PR DESCRIPTION
**Pull request by minsiyang**
- [x] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [ ] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [x] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does

Sanitize workload metadata before sending it to K-U

### CODE REVIEW

File: test/fixtures/sidecar-containers/deployment.yaml [hunk 1]
```diff
@@ -48,6 +48,9 @@ spec:
           successThreshold: 1
           timeoutSeconds: 5
         name: hello-world
+        args: [something]
```
Comment (dkontorovskyy) [inline test/fixtures/sidecar-containers/deployment.yaml:51]: Let's add `env` also?

Should we have an explicit test to validate that this key is not being sent?

**Comment by dkontorovskyy** [comment]
AMAZING 👑

File: test/fixtures/sidecar-containers/deployment.yaml (hunk 1 above)
Comment (minsiyang) [reply test/fixtures/sidecar-containers/deployment.yaml:51]: the unit test of `metadata-extractor.spec.ts` should validate that?

Comment (minsiyang) [reply test/fixtures/sidecar-containers/deployment.yaml:51]: I see what you mean now, sorry. I will add a test now 😅

**Pull request merged**